addons/EasyNuke/EasyNuke.lua

```diff
@@ -0,0 +1,207 @@
+
+require('chat')
+require('lists')
+require('logger')
+require('sets')
+require('tables')
+require('strings')
+require('pack')
+
+file  = require('files')
+config = require('config')
+texts = require('texts')
+res = require('resources')
+
+local defaults = T{}
+defaults.current_element = "fire"
+defaults.target_mode = "t"
+settings = config.load(defaults)
+
+current_element = "fire"
+target_mode = "t"
+
+elements = T{"fire","wind","thunder","light","ice","water","earth","dark"}
+elements_dark = T{"ice","water","earth","dark"}
+elements_light = T{"fire","wind","thunder","light"}
+elements_index = 1
+other_modes = T{"fire","wind","thunder","light","ice","water","earth","dark","drain","aspir","absorb","cure"}
+
+targets = T{"t","bt","stnpc"}
+targets_index = 1
+
+spell_tables = {}
+spell_tables["fire"] = T{"Fire","Fire II","Fire III","Fire IV","Fire V","Fire VI",}
+spell_tables["fire"]["ga"] = T{"Firaga","Firaga II","Firaga III","Firaja",}
+spell_tables["fire"]["ra"] = T{"Fira","Fira II","Fira III"}
+spell_tables["earth"] = T{"Stone","Stone II","Stone III","Stone IV","Stone V","Stone VI",}
+spell_tables["earth"]["ga"] = T{"Stonega","Stonega II","Stonega III","Stoneja",}
+spell_tables["earth"]["ra"] = T{"Stonera","Stonera II","Stonera III"}
+spell_tables["wind"] = T{"Aero","Aero II","Aero III","Aero IV","Aero V","Aero VI",}
+spell_tables["wind"]["ga"] = T{"Aeroga","Aeroga II","Aeroga III","Aeroja",}
+spell_tables["wind"]["ra"] = T{"Aerora","Aerora II","Aerora III"}
+spell_tables["water"] = T{"Water","Water II","Water III","Water IV","Water V","Water VI",}
+spell_tables["water"]["ga"] = T{"Waterga","Waterga II","Waterga III","Waterja",}
+spell_tables["water"]["ra"] = T{"Watera","Watera II","Watera III"}
+spell_tables["ice"] = T{"Blizzard","Blizzard II","Blizzard III","Blizzard IV","Blizzard V","Blizzard VI",}
+spell_tables["ice"]["ga"] = T{"Blizzaga","Blizzaga II","Blizzaga III","Blizzaja",}
+spell_tables["ice"]["ra"] = T{"Blizzara","Blizzara II","Blizzara III"}
+spell_tables["thunder"] = T{"Thunder","Thunder II","Thunder III","Thunder IV","Thunder V","Thunder VI",}
+spell_tables["thunder"]["ga"] = T{"Thundaga","Thundaga II","Thundaga III","Thundaja",}
+spell_tables["thunder"]["ra"] = T{"Thundara","Thundara II","Thundara III"}
+spell_tables["light"] = T{"Banish","Banish II","Holy","Banish III",}
+spell_tables["light"]["ga"] = T{"Banishga","Banishga II"}
+spell_tables["dark"] = T{"Impact"}
+spell_tables["dark"]["ga"] = T{"Comet"}
+spell_tables["cure"] = T{"Cure","Cure II","Cure III","Cure IV","Cure V","Cure VI"}
+spell_tables["cure"]["ga"] = T{"Curaga","Curaga II","Curaga III","Curaga IV","Curaga V",}
+spell_tables["cure"]["ra"] = T{"Cura","Cura II","Cura III"} 
+spell_tables["drain"] = T{"Aspir","Aspir II","Aspir III","Drain","Drain II","Drain III"}
+spell_tables["absorb"] = T{"Absorb-Acc","Absorb-TP","Absorb-Attri","Absorb-STR","Absorb-DEX","Absorb-VIT","Absorb-AGI","Absorb-INT","Absorb-MND","Absorb-CHR"}
+
+function find_index ()
```

Comment:
This function could be replaced with a lookup table:
```lua
local indices = {
    fire = 1,
    wind = 2,
    thunder = 3,
    light = 4,
    ice = 5,
    water = 6,
    earth = 7,
    dark = 8,
}
```

Then you can just use `indices[current_element]` later.

Reply:
Working on this now.  Since it's going to change the "cycle" logic, might take a little bit to wrap my head around what needs to change first tho.